lib/pure/smtp.nim

```diff
@@ -141,31 +153,92 @@ proc close*(smtp: Smtp) =
   smtp.debugSend("QUIT\c\L")
   smtp.sock.close()
 
-proc createMessage*(mSubject, mBody: string, mTo, mCc: seq[string],
+
+proc getMimeTypeOfFile(pathToFile:string): string =
+  ## This will return the mimetype
+  var mimeDB = newMimetypes()
+  var ext = extractExt(pathToFile) # 
+
+  ## we choose "application/octet-stream" 
+  ## if no extension is given
+  ## TODO what about linux? or files without extension.
+  ## Do we have to have something like `file` magic?
+  return mimeDB.getMimetype(ext,"bin") 
```

Comment:
I think this is good as a default heuristic for determining the mimetype. What I would like you to do is to overload the ``attach`` procedure so that the user can optionally specify the mimetype, something like this:

```nim
proc attach*(msg: var Message, filepath: string, msgFilename: string) = ...
proc attach*(msg: var Message, filepath: string, msgFilename: string, mimetype: string) = ...
```